cpp/test/sg/fil_test.cu

```diff
@@ -518,13 +564,24 @@ class TreeliteFilTest : public BaseFilTest {
           builder->SetLeafNode(
             key, tlf::Value::Create(dense_node.base_node::output<val_t>().f));
           break;
-        case fil::leaf_algo_t::CATEGORICAL_LEAF:
+        case fil::leaf_algo_t::CATEGORICAL_LEAF: {
           std::vector<tlf::Value> vec(ps.num_classes);
           for (int i = 0; i < ps.num_classes; ++i) {
             vec[i] = tlf::Value::Create(
               i == dense_node.base_node::output<val_t>().idx ? 1.0f : 0.0f);
           }
           builder->SetLeafVectorNode(key, vec);
+          break;
+        }
+        case fil::leaf_algo_t::VECTOR_LEAF: {
+          std::vector<tlf::Value> vec(ps.num_classes);
+          for (int i = 0; i < ps.num_classes; ++i) {
+            auto idx = dense_node.base_node::output<val_t>().idx;
```

Comment:
I wonder why we used `<val_t>().idx` as opposed to `<int>()` straight away...
Also, could you change the `.base_node::output` to `template dense_node.output`? Feel free to fix earlier as well.
I didn't add as patch suggestion because I need my cmake fixed before I can test